### PR TITLE
Pass GPG Directory as Parameter to templates

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Resolves https://github.com/Azure/azure-sdk-for-java/issues/10196
Depends on:

1. https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools/pullrequest/200
2. Run of https://dev.azure.com/azure-sdk/internal/_build?definitionId=1698
3. Release of `azure-sdk-template`

### CODE REVIEW

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 1]
```diff
@@ -348,6 +348,8 @@ stages:
           timeoutInMinutes: 5
 
         - template: tools/gpg/gpg.yml@azure-sdk-build-tools
+          parameters:
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:351]: We should default this in the gpg.yml template given it is in the azure-sdk-build-tools repot.

File: eng/pipelines/templates/steps/java-publishing.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@ parameters:
   RepositoryUrl: not-specified
   GroupID:
   ArtifactID:
-  BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
+  GPGToolDirectory: ''
```
Comment (weshaggard) [inline eng/pipelines/templates/steps/java-publishing.yml:6]: Probably still worth defaulting this here.

File: eng/pipelines/templates/stages/archetype-java-release.yml [hunk 3]
```diff
@@ -358,5 +360,5 @@ stages:
                 ArtifactDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed
                 RepositoryUrl: $(RepositoryUrl)
                 Target: JavaDevFeed
-                BuildToolsPath: $(Pipeline.Workspace)/azure-sdk-build-tools
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/archetype-java-release.yml:361]: Lets just remove this and depend on the default value.

**Comment by hallipr** [comment]
Do we do anything else with `$GPGToolDirectory` other than set `$GPGExecutablePath = $GPGToolDirectory\gpg.exe`?  Or, should we just replace BuildToolsPath with its implicit GpgExecutablePath with an explicit GPGExecutablePath instead of replacing one implicit path with another?

Is it possible that someone won't have gpg.exe in $GPGToolDirectory and would prefer to use `$GPGExecutablePath = 'some/path/to/gpg-1.2.3.beta.exe'

**Review by hallipr — APPROVED** [review]

**Comment by chidozieononiwu** [comment]
> Do we do anything else with `$GPGToolDirectory` other than set `$GPGExecutablePath = $GPGToolDirectory\gpg.exe`? Or, should we just replace `$BuildToolsPath` and its implicit `$GPGExecutablePath` with an explicit `$GPGExecutablePath` instead of replacing one implicit path with another?
> 
> Is it possible that someone won't have gpg.exe in $GPGToolDirectory and would prefer to use `$GPGExecutablePath = 'some/path/to/gpg-1.2.3.beta.exe'`?

I have updated that. Please also take a look at https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-build-tools/pullrequest/200

**Review by weshaggard — APPROVED** [review]

**Comment by chidozieononiwu** [comment]
Tested with this build https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1081774&view=results

**Pull request merged**